src/paint.js

```diff
@@ -169,31 +167,30 @@ function createDataArray(hypercubeData, layout) {
 }
 
 export default function paint($element, layout, component) {
-
   //set hypercube variable and call function on hcData to return data in a json format
-  var hc = layout.qHyperCube,
+  let hc = layout.qHyperCube,
     hcData = createDataArray(hc, layout);
 
   //create variables for number of bars allowed and the size of the dimension area for text
-  var dimWidth = Number(layout.props.section1.dimWidth),
+  let dimWidth = Number(layout.props.section1.dimWidth),
     barsNum = layout.props.section2.barNum;
 
 
   // Create margin - should be replaced by dynamic numbers when this is eventually a responsive viz
-  var margin = { top: 5, right: 20, bottom: 25, left: dimWidth };
+  let margin = { top: 5, right: 20, bottom: 25, left: dimWidth };
 
   // Set chart object width
-  var width = $element.width() - margin.left - margin.right;
+  let width = $element.width() - margin.left - margin.right;
 
   // Set chart object height
   if(barsNum > hcData.length){
     barsNum = hcData.length;
   }
-  var height =Math.abs($element.height() / barsNum - margin.top - margin.bottom - 1);
+  let height =Math.abs($element.height() / barsNum - margin.top - margin.bottom - 1);
 
   // Chart object id
-  var id = 'container_' + layout.qInfo.qId;
-
+  const randomId = component.definitions.randomId;
+  let id = 'container_' + layout.qInfo.qId + randomId;
   // Check to see if the chart element has already been created
   if (document.getElementById(id)) {
```

Comment:
You shouldn't need a unique ID at all if we just switch to using $element.getElementById instead of searching through the entire document?